### DISCUSSION TITLE
Multiple Slingboxes on Linux (Debian)

### DISCUSSION THREAD

**Discussion by tom76017** [GerryDazoo/Slinger#27] [Q&A] [answered]
Will this work on my Linux server to support a SB500 and M1 from on 1 server?

**_sling.service_**

[Unit]
Description=SlingBox Server Service
After=multi-user.target

[Service]
Type=idle
User=slingbox
WorkingDirectory=/home/slingbox
StandardOutput=append:/tmp/sling.log
StandardError=append:/tmp/sling.log
**ExecStart=/usr/bin/python3 -u slingbox_server.py config-SB500.ini config-M1.ini** 
Restart=always

[Install]
WantedBy=multi-user.target

**Comment by GerryDazoo** [comment, marked as answer]
The short is it “should” depending of which version of system is running on your box. You may find the log file isn’t being updated.

 

I’ve had to adjust the service config for older systemd’s. Something like this

 

[Unit]

Description=SlingBox Server Service

After=multi-user.target

 

[Service]

Type=idle

User=slingbox

WorkingDirectory=/home/slingbox

ExecStart=/bin/bash -ce "exec /usr/bin/python3 -u slingbox_server.py 500_config.ini 500lbr_config.ini >> /tmp/sling.log 2>&1"

Restart=always

 

[Install]

WantedBy=multi-user.target

 

From: tom76017 ***@***.***> 
Sent: Wednesday, August 17, 2022 3:10 PM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] Multiple Slingboxes on Linux (Debian) (Discussion #27)

 

Will this work on my Linux server to support a SB500 and M1 from on 1 server?

[Unit]
Description=SlingBox Server Service
After=multi-user.target

[Service]
Type=idle
User=slingbox
WorkingDirectory=/home/slingbox
StandardOutput=append:/tmp/sling.log
StandardError=append:/tmp/sling.log
ExecStart=/usr/bin/python3 -u slingbox_server.py config-SB500.ini config-M1.ini
Restart=always

[Install]
WantedBy=multi-user.target

**Comment by tom76017** [comment]
Works on Debian with 2 500's and 1 M1.  I do get an occasional broken pipe error.  Th remote works fine with both my TiVo Mini A92 and A95's.  Just a little slow.

BTW, I noticed the a hard reset changes the password and finderid.  I wonder what will happen with the passwod and finderid with a hard reset after EOL.

**Comment by GerryDazoo** [comment]
That’s good news. Can you can you cut and paste the lines around the broken pipe error for me?

Make sure you’ve updated the  system TCP buffer sizes as described in the readme.

It’s not the remote that’s slow (check it while watching the slingbox input directly). It’s video stream buffering, etc in your player.  I’m assuming you’re talking about VLC. You can google up how to minimize the delays but it’s a pain.

I’ve been using ffplayer with some success. With the proper command line the delays make the remote usable.

i.e.  C:\ffmpeg\ffplay.exe -infbuf -probesize 32 -analyzeduration 0  -flags nobuffer -flags low_delay http://127.0.0.1:65432/slingbox

 

Yes a hard reset generates a new finderid and password. I would NOT do a hard reset after EOL. I’m not brave enough to give it a test but if you want to see what might happen you could block internet access to your PC. I suspect it would fail miserably and leave your box in an unusable state because it can’t get a new password and finderid from Mr. Sling.

After the test you could re-enable Internet access and should be able to recover the slingbox.

 

*	Gerry

**Reply by GerryDazoo** [reply]
Based on my testing ( I used the firewall to block all outgoing connections to the WAN), do NOT do a hard reset after EOL. The sling software will timeout trying to talk to the cloud and there will be no way to recover/set the password.
